frontend/src/components/pages/camps/CampersTable.tsx

```diff
@@ -0,0 +1,190 @@
+
+import {
+  Container,
+  Table,
+  Text,
+  Thead,
+  Tbody,
+  Tr,
+  Td,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  VStack,
+  IconButton,
+  HStack,
+  Button,
+} from "@chakra-ui/react";
+import { DownloadIcon, SearchIcon } from "@chakra-ui/icons";
+
+import textStyles from "../../../theme/textStyles";
+import { Camper } from "../../../types/CamperTypes";
+
+import CampersTableOverviewIconGroup from "./CampersTableOverviewIconGroup";
+import { CamperDetailsBadgeGroup } from "./CamperDetailsBadges";
+
+const ExportButton = () => {
+  return (
+    <Button
+      leftIcon={<DownloadIcon />}
+      aria-label="Export SVG"
+      type="submit"
+      background="background.grey.200"
+      color="primary.green.100"
+      border="2px"
+      borderColor="primary.green.100"
+      borderRadius="5px"
+      minWidth="10vw"
+    >
+      Export as .csv
+    </Button>
+  );
+};
+
+const CampersTable = ({
+  campers,
+  campCapacity,
+}: {
+  campers: Camper[];
+  campCapacity: number;
+}) => {
+  const [displayedCampers, setDisplayedCampers] = React.useState(campers);
+  const [search, setSearch] = React.useState("");
+
+  const tableData = React.useMemo(() => {
+    const filteredCampers = displayedCampers;
+
+    if (!search) return filteredCampers;
+    return filteredCampers.filter((camper: any) =>
+      camper.firstName
+        .toLowerCase()
+        .concat(" ", camper.lastName.toLowerCase())
+        .includes(search.toLowerCase()),
+    );
+  }, [search, displayedCampers]);
+
+  const camperDetailsCount: { [key: string]: number } = {};
+  campers.forEach((camper) => {
+    if (camper.earlyDropoff) {
+      if (camperDetailsCount.earlyDropoff) camperDetailsCount.earlyDropoff += 1;
+      else camperDetailsCount.earlyDropoff = 1;
+    }
+    if (camper.latePickup) {
+      if (camperDetailsCount.latePickup) camperDetailsCount.latePickup += 1;
+      else camperDetailsCount.latePickup = 1;
+    }
+    if (camper.allergies) {
+      if (camperDetailsCount.allergies) camperDetailsCount.allergies += 1;
+      else camperDetailsCount.allergies = 1;
+    }
+    if (camper.specialNeeds) {
+      if (camperDetailsCount.specialNeeds) camperDetailsCount.specialNeeds += 1;
+      else camperDetailsCount.specialNeeds = 1;
+    }
+  });
+
+  return (
+    <Container
+      maxWidth="90vw"
+      px="-5"
+      py="5"
+      background="background.grey.200"
+      borderRadius="20"
+    >
+      {campers.length > 0 ? (
+        <>
+          <HStack spacing={12} px="18">
+            <InputGroup>
+              <InputLeftElement pointerEvents="none">
+                <SearchIcon color="gray.300" />
+              </InputLeftElement>
+              <Input
+                size="md"
+                enterKeyHint="enter"
+                onChange={(e) => setSearch(e.target.value)}
+                placeholder="Search by camper name..."
+              />
+            </InputGroup>
+            <CampersTableOverviewIconGroup
```

Comment:
The icon buttons above the camper rows are meant to be filters and the counts in each of these filters needs to be dynamically changing based on the search results. I think you would need to pass in `tableData` to the icons so that you can keep updating the numbers in the icons as the user searches in the search bar. You will also need a way to filter results when a user clicks on the icon buttons. As a result, the counts of the other filters will change accordingly. 

<img width="1289" alt="image" src="https://user-images.githubusercontent.com/57159970/186125339-e9a230fb-148e-4c71-bfa1-a1719b4f0997.png">
All the filters besides the ( person 4/20) filter should be updated with the new counts upon search